M2/Macaulay2/m2/programs.m2

```diff
@@ -42,15 +50,26 @@ getProgramPath = (name, cmds, opts) -> (
 	    if opts.Verbose == true and #prefixes > 1 then
 		print("  trying prefix \"" | prefix_1 |
 		    "\" for executables matching \"" | prefix_0 | "\"...");
+	    newOpts := if opts.MinimumVersion =!= null then
```

Comment:
Why not just assign the new value to `opts`?

Reply:
The new value contains the path and prefix information specific to the particular iteration we're working on, so if we use `opts`, then we'd lose the original value without that stuff.

For example, let's say we want to find the program `foo` with version at least 1.0.  And maybe some Linux distribution prepends all the various components of the `foo` program with `bar-`.  We might call something like:
```m2
findProgram("foo", "foo --help", Prefix => (".*", "bar-"), MinimumVersion => ("1.0", "foo --version"))
```

But to check the version number for a particular path/prefix combo, we want to insert that information into the `MinimumVersion` sequence.  So when checking, say, `/usr/local/bin` with the `bar-` prefix, we'd want `MinimumVersion` to be `("1.0", "/usr/local/bin/bar-foo --version")`.  That's what `newOpts` contains.  It doesn't remember what the original sequence in `opts` looks like, which is why I went with a new variable.

Looking at the code again, I think it may be simpler to just send the prefix and path information to the `checkProgramPath` function and let it do everything.